sdk/core/azure-core/azure/core/_pipeline_client.py

```diff
@@ -131,6 +132,7 @@ def _build_pipeline(self, config, **kwargs):  # pylint: disable=no-self-use
                 [
                     config.logging_policy,
                     DistributedTracingPolicy(**kwargs),
+                    SensitiveHeaderCleanupPolicy(**kwargs),
```

Comment:
If we start generating SDKs without a redirect policy - what will this do?

Reply:
Because `insecure_domain_change` is not set, it does nothing.

Reply:
In other words, if we decide to stop generating redirect policy, we should stop adding cleanup policy too.

Reply:
With the current implementation, if we stop generating the redirect policy - no changes should be needed in Core, because we already [filter out](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/_base.py#L136) `None` policies.

Reply:
If redirect policy is removed, it looks to me there is almost no overhead added in the cleanup policy. https://github.com/Azure/azure-sdk-for-python/blob/3058d921feb59cfadc3653ba38e961d07b7afdd1/sdk/core/azure-core/azure/core/pipeline/policies/_sensitive_header_cleanup_policy.py#L58-L59

Reply:
The presence of the policy is overhead. I would prefer something like:
`SensitiveHeaderCleanupPolicy(**kwargs) if config.redirect_policy else None`

This will be "self cleaning" in the case that a generated library didn't have the redirect policy.